compose/network.py

```diff
@@ -159,18 +159,29 @@ def initialize(self):
             network.ensure()
 
 
-def get_network_names_for_service(service_dict):
+def get_network_aliases_for_service(service_dict):
     if 'network_mode' in service_dict:
-        return []
-    return service_dict.get('networks', ['default'])
+        return {}
+    networks = service_dict.get('networks', ['default'])
+    if isinstance(networks, list):
+        return dict((net, []) for net in networks)
+
+    return dict(
+        (net, (config or {}).get('aliases', []))
+        for net, config in networks.items()
+    )
```

Comment:
I would actually expect that this normalization would need to happen in `compose/config/config.py`, otherwise merging networks from multiple files will probably fail when one is a list and the other is a dict.

Mind adding a test to `tests/unit/config/config_test.py` to confirm ?